x-pack/plugins/ml/server/models/job_service/datafeeds.ts

```diff
@@ -168,38 +170,62 @@ export function datafeedsProvider(client: IScopedClusterClient, mlClient: MlClie
     }, {} as { [id: string]: string });
   }
 
+  async function getDatafeedByJobId(
+    jobId: string[],
+    excludeGenerated?: boolean
+  ): Promise<Datafeed[] | undefined>;
+
   async function getDatafeedByJobId(
     jobId: string,
     excludeGenerated?: boolean
-  ): Promise<Datafeed | undefined> {
+  ): Promise<Datafeed | undefined>;
+
+  async function getDatafeedByJobId(
+    jobId: string | string[],
+    excludeGenerated?: boolean
+  ): Promise<Datafeed | Datafeed[] | undefined> {
+    const jobIds = Array.isArray(jobId) ? jobId : [jobId];
+
     async function findDatafeed() {
       // if the job was doesn't use the standard datafeedId format
       // get all the datafeeds and match it with the jobId
       const {
         body: { datafeeds },
-      } = await mlClient.getDatafeeds(excludeGenerated ? { exclude_generated: true } : {}); //
-      for (const result of datafeeds) {
-        if (result.job_id === jobId) {
-          return result;
-        }
+      } = await mlClient.getDatafeeds(excludeGenerated ? { exclude_generated: true } : {});
+      if (typeof jobId === 'string') {
+        return datafeeds.find((v) => v.job_id === jobId);
+      }
+
+      if (Array.isArray(jobId)) {
+        return datafeeds.filter((v) => jobIds.includes(v.job_id));
       }
     }
     // if the job was created by the wizard,
     // then we can assume it uses the standard format of the datafeedId
-    const assumedDefaultDatafeedId = `datafeed-${jobId}`;
+    const assumedDefaultDatafeedId = jobIds.map((v) => `datafeed-${v}`).join(',');
     try {
       const {
         body: { datafeeds: datafeedsResults },
       } = await mlClient.getDatafeeds({
         datafeed_id: assumedDefaultDatafeedId,
         ...(excludeGenerated ? { exclude_generated: true } : {}),
       });
-      if (
-        Array.isArray(datafeedsResults) &&
-        datafeedsResults.length === 1 &&
-        datafeedsResults[0].job_id === jobId
-      ) {
-        return datafeedsResults[0];
+      if (Array.isArray(datafeedsResults)) {
+        const result = datafeedsResults.filter((d) => jobIds.includes(d.job_id));
```

Comment:
this line could be moved down a few lines so `result` is only created just before being used.